clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/callback/PlaintextConsumerCallbackTest.java

```diff
@@ -0,0 +1,382 @@
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.test.ClusterInstance;
+import org.apache.kafka.common.test.TestUtils;
+import org.apache.kafka.common.test.api.ClusterTest;
+import org.apache.kafka.common.test.api.ClusterTestDefaults;
+import org.apache.kafka.common.test.api.Type;
+import org.apache.kafka.common.test.junit.ClusterTestExtensions;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_PROTOCOL_CONFIG;
+import static org.apache.kafka.clients.consumer.GroupProtocol.CLASSIC;
+import static org.apache.kafka.clients.consumer.GroupProtocol.CONSUMER;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ClusterTestDefaults(
+    types = {Type.KRAFT},
+    brokers = 3
+)
+@ExtendWith(ClusterTestExtensions.class)
+public class PlaintextConsumerCallbackTest {
+
+    private final ClusterInstance cluster;
+    private final String topic = "topic";
+    private final TopicPartition tp = new TopicPartition(topic, 0);
+
+    public PlaintextConsumerCallbackTest(ClusterInstance clusterInstance) {
+        this.cluster = clusterInstance;
+    }
+
+    @ClusterTest
+    public void testConsumerRebalanceListenerAssignOnPartitionsAssigned() throws InterruptedException {
+        try (var consumer = createConsumer(CLASSIC)) {
+            triggerOnPartitionsAssigned(tp, consumer, (executeConsumer, partitions) -> {
+                var e = assertThrows(IllegalStateException.class, () -> executeConsumer.assign(List.of(tp)));
+                assertEquals("Subscription to topics, partitions and pattern are mutually exclusive", e.getMessage());
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerRebalanceListenerAssignOnPartitionsAssigned() throws InterruptedException {
+        try (var consumer = createConsumer(CONSUMER)) {
+            triggerOnPartitionsAssigned(tp, consumer, (executeConsumer, partitions) -> {
+                var e = assertThrows(IllegalStateException.class, () -> executeConsumer.assign(List.of(tp)));
+                assertEquals("Subscription to topics, partitions and pattern are mutually exclusive", e.getMessage());
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testConsumerRebalanceListenerAssignmentOnPartitionsAssigned() throws InterruptedException {
```

Comment:
You have `testAsyncConsumer...` above, so I suggest you also name it `testClassicConsumer...` also